point-of-sale-app/inventory/src/main/java/com/google/abmedge/inventory/dao/InMemoryStoreConnector.java

```diff
@@ -0,0 +1,130 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.abmedge.inventory.dao;
+
+import com.google.abmedge.inventory.dto.Item;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * An implementation of the {@link InventoryStoreConnector} that uses an in memory data structure to
+ * hold all the inventory information. This implementation uses a simple HashMap to store all the
+ * items
+ */
+public class InMemoryStoreConnector implements InventoryStoreConnector {
+
+  private static final Logger LOGGER = LogManager.getLogger(InMemoryStoreConnector.class);
+  private static final Map<UUID, Item> idToItemsMap = new HashMap<>();
+
+  @Override
+  public List<Item> getAll() {
+    return idToItemsMap.values()
```

Comment:
Are these items really big that we can't do toArray() directly without invoking stream(..) API? The Java streams are a great abstraction for MR like operations but over here this looks bit overkill.

Reply:
I am sure I understand. The `toArray` method against what object? Only reason I am stream processing each item and collecting into a list is because I need to take a deep-copy of the item before returning to ensure that mutation of it outside this API doesn't effect the underlying map.